lua/notes.lua

```diff
@@ -0,0 +1,111 @@
+local M = {}
+
+-- Open a note file, automatically appending .md extension if not present
+-- Creates file with header if it doesn't exist
+function M.notes_open(filename)
+  -- Check if filename already ends with .md
+  local final_filename
+  if not filename:match('%.md$') then
+    final_filename = filename .. '.md'
+  else
+    final_filename = filename
+  end
+
+  -- Check if the file exists, if not, create it with a header
+  if vim.fn.filereadable(final_filename) == 0 then
+    -- Extract note name from filename (remove .md extension)
+    local note_name = final_filename:gsub('%.md$', '')
+    local header = '# ' .. note_name
+    vim.fn.writefile({header}, final_filename)
+  end
+
+  vim.cmd('edit ' .. final_filename)
+end
+
+-- Treats the current line as a link and open that file
+-- First it looks for obsidian style links
+-- Otherwise it looks for items in the format '- Item'
+function M.open_current()
+  local line = vim.fn.getline('.')
+  local cursor_col = vim.fn.col('.')
+  local pattern = '%[%[(.-)%]%]'
+  local matches = {}
+
+  -- Find all the matches for links
+  local start = 1
+  while true do
+    local match_start, match_end, match_text = line:find(pattern, start)
+    if not match_start then
+      break
+    end
+    table.insert(matches, {
+      text = '[[' .. match_text .. ']]',
+      pos = match_start,
+      inner_text = match_text
+    })
+    start = match_end + 1
+  end
+
+  local filename
```

Comment:
check if this variable should be renamed to `title` too

Reply:
it isn't the filename since it doesn't include `.md` and it is passed as the `title`! so it should be `title` here too.